### PR TITLE
Cc shim bug fixes

### PR DESCRIPTION
**Pull request by amshinde**

### CODE REVIEW

File: shim/shim.c [hunk 1]
```diff
@@ -238,7 +241,11 @@ handle_signals(char *container_id, int outfd) {
 	while (read(signal_pipe_fd[0], &sig, sizeof(sig)) != -1) {
 		printf("Handling signal : %d on fd %d\n", sig, signal_pipe_fd[0]);
 		if (sig == SIGWINCH ) {
-			cmd_type = WINSIZE;
+			cmd = strdup("winsize");
```
Comment (dlespiau) [inline shim/shim.c]: No need for the underlying strlen + allocation here just define 2 static variables (static const char[]) and assign cmd to the right one.

File: shim/shim.c [hunk 2]
```diff
@@ -169,7 +171,7 @@ get_proxy_ctl_msg(char *json, size_t *len) {
  * \param json Json payload
  */
 void
-send_proxy_hyper_message(int fd, int hyper_cmd_type, char *json) {
+send_proxy_hyper_message(int fd, char *hyper_cmd, char *json) {
```
Comment (dlespiau) [inline shim/shim.c]: probably a good time to start sprinkling consts :)

const char *hyper_cmd

File: shim/shim.c [hunk 3]
```diff
@@ -148,14 +149,15 @@ get_proxy_ctl_msg(char *json, size_t *len) {
 		return NULL;
 	}
 
-	*len = strlen(json) + PROXY_CTL_HEADER_SIZE + 1;
-	proxy_ctl_msg = calloc(*len, sizeof(char));
+	*len = strlen(json) + PROXY_CTL_HEADER_SIZE;
```
Comment (dlespiau) [inline shim/shim.c]: Could we only have one invocation of strlen(json) while you're here?

**Comment by amshinde** [comment]
@dlespiau Have pushed all the changes you mentioned.

**Comment by dlespiau** [comment]
lgtm

**Comment by jodh-intel** [comment]
lgtm.

**Comment by chavafg** [comment]
qa-failed

**Pull request merged**